app/mailers/notification_mailer.rb

```diff
@@ -371,7 +371,7 @@ def weekly_report(report, receiver)
   # @param [User] received
   delivers Services::ServiceDeletedEvent
   def service_deleted(event, receiver)
-    @provider_account = event.provider
+    @provider_account = Account.find(event.metadata[:provider_id])
```

Comment:
Can the account no longer exist at this point? Won't we be raising a `RecordNotFound` in this case?

Reply:
@guicassolato If the provider does not exist, it shouldn't even reach here 😉 If it reaches here and it doesn't exist, then indeed it should raise an error. Why would we want to send an email to accounts that don't exist anymore because a service has been deleted? 😄 
But `NotificationEvent` is not created if the `provider` attribute does not exist in `Services::ServiceDeletedEvent` (which is what indicates if the provider still exists or not) and then it shouldn't reach here 😄

Reply:
I was concerned about https://github.com/3scale/porta/blob/46eb05a4513e21271498d887c9e55514eba64c85/app/events/services/service_deleted_event.rb#L5. It seems that `metadata.provider_id` can come from an account that is built but not persisted.